filesmanager/filesmanager.py

```diff
@@ -48,20 +121,344 @@ def student_view(self, context=None):
+
+        return Response(
+            status=HTTPStatus.OK,
+            json_body=target_directory,
+        )
+
     @XBlock.json_handler
-    def increment_count(self, data, suffix=''):
+    def reorganize_content(self, data, suffix=''):
+        """Reorganize a content from a source path to a target path.
+
+        Arguments:
+            target_path: the path of the target directory where the content will be moved.
+            source_path: the path of the source directory where the content is located.
+            target_index: the index where the content will be inserted in the target directory.
+
+        Returns: the content of the target directory.
+        """
+        target_path = data.get("target_path")
+        source_path = data.get("source_path")
+        target_index = data.get("target_index")
+        if not target_path or not source_path:
+            return {
+                "status": "error",
+                "message": "Path not found",
+            }
+        content, index, source_parent_directory = self.get_content_by_path(source_path)
+        target_content, _, target_parent_directory = self.get_content_by_path(target_path)
+        if not content or not target_content:
+            return {
+                "status": "error",
+                "message": "Content not found",
+            }
+        new_content_path = f"{target_content['path']}/{content['name']}"
+        del source_parent_directory[index]
+        if target_index is None:
+            target_content["children"].append(content)
+        else:
+            target_index = 0 if not target_index else int(target_index)
+            new_target_content_directory_children = target_content["children"][:target_index]
+            new_target_content_directory_children.append(content)
+            new_target_content_directory_children.extend(target_content["children"][target_index:])
+            target_content["children"] = new_target_content_directory_children
+        self.update_content_path(content, new_content_path)
+        return {
+            "status": "success",
+            "content": target_parent_directory,
+        }
+
+    @XBlock.json_handler
+    def delete_content(self, data, suffix=''):
+        """Delete a content from the course assets.
+
+        Arguments:
+            path: the path of the content to be deleted.
+
+        Returns: the content of the parent directory.
+        """
+        paths = data.get("paths")
+        if not paths:
+            return {
+                "status": "error",
+                "message": "Path not found",
+            }
+        for path in paths:
+            content, index, parent_directory = self.get_content_by_path(path)
+            if content:
+                del parent_directory[index]
+                self.delete_content_from_assets(content)
+        return {
+            "status": "success",
+        }
+
+    def get_target_directory(self, path):
```

Comment:
I think we should add type hints in the functions, this way we have clarity about the types it receives and the types it returns.

Reply:
I'll consider this a nit